zbuf/reader.go

```diff
@@ -0,0 +1,80 @@
+package zbuf
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/zio"
+)
+
+type Control struct {
+	Message interface{}
+}
+
+var _ error = (*Control)(nil)
+
+func (c *Control) Error() string {
+	return "control"
+}
+
+type SetChannel int
+type EndChannel int
+
+type noControl struct {
+	zio.Reader
+}
+
+func NoControl(r zio.Reader) *noControl {
+	return &noControl{Reader: r}
+}
+
+func (n *noControl) Read() (*zed.Value, error) {
+	for {
+		val, err := n.Reader.Read()
+		if err != nil {
+			if _, ok := err.(*Control); ok {
+				continue
+			}
+		}
+		return val, err
+	}
+}
+
+type ProgressReader interface {
+	zio.Reader
+	Progress() Progress
+}
+
+type meterReader struct {
+	zio.Reader
+	progress Progress
+}
+
+var _ ProgressReader = (*meterReader)(nil)
+
+func MeterReader(r zio.Reader) *meterReader {
+	return &meterReader{Reader: r}
+}
```

Comment:
```suggestion
```suggestion
func MeterReader(r zio.Reader) ProgressReader {
	return &meterReader{Reader: r}
}

type meterReader struct {
	zio.Reader
	progress Progress
}
```
```